pkg/operators/wasm/filter.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import (
+	"context"
+
+	"github.com/tetratelabs/wazero"
+	wapi "github.com/tetratelabs/wazero/api"
+)
+
+func (i *wasmOperatorInstance) addFilterFuncs(env wazero.HostModuleBuilder) {
+	exportFunction(env, "shouldDiscardMntnsID", i.shouldDiscardMntnsID,
+		[]wapi.ValueType{
+			wapi.ValueTypeI64, // MntNS ID
+		},
+		[]wapi.ValueType{wapi.ValueTypeI32}, // Discard the event, 1 if true, 0 if false
+	)
+}
+
+// shouldDiscardMntnsID returns 1 if the mount ns ID should be filtered out.
+// Params:
+// - stack[0]: mount ns ID
+// Return value:
+// - 1 if the mount ns ID should be filter out, 0 otherwise
+func (i *wasmOperatorInstance) shouldDiscardMntnsID(ctx context.Context, m wapi.Module, stack []uint64) {
+	// if the filtering map is not configured, we assume the gadget wants to get
+	// all data
+	if i.mntNsIDMap == nil {
+		stack[0] = 0
+		return
```

Comment:
Even if the advise seccomp gadget does not include "include/gadget/mntns_filter.h", the MntNsFilterMapName map is created by localmanager or kubemanager, so it works. But this seems fragile to me, it could stop working if the localmanager or kubemanager was implemented differently.

Reply:
You're right. I explicitly imported the file on the gadget to avoid it relying on {local,kube}manager implementations.